addons/website_sale_slides/models/slide_channel.py

```diff
@@ -70,3 +71,24 @@ def _filter_add_members(self, target_partners, **member_values):
             else:
                 result |= on_payment
         return result
+
+    def default_get(self, default_fields):
+        values = super(Channel, self).default_get(default_fields)
+
+        if 'product_id' in default_fields:
+            product_courses = self.env['product.product'].search(
+                [('detailed_type', '=', 'course')], limit=2)
+            if len(product_courses) == 1:
+                values['product_id'] = product_courses.id
+
+        return values
+
+    @api.model
+    def get_course_products(self):
```

Comment:
I would put it into a controller and behind a (technical) route as basically it has nothing to do with channel model.